File: 剑指 Offer 05. 替换空格/replacespace.java
// Java Python 的字符串被设计成不可变类型
class Solution {
    public String replaceSpace(String s) {
        StringBuilder res = new StringBuilder();
        for (Character c : s.toCharArray())
        {
            if(c == ' ') res.append("%20");
            else res.append(c);
        }
        return res.toString();
    }
}

File: 剑指 Offer 05. 替换空格/replacespace.go
package main

func replaceSpace(s string) string {
	b := []byte(s)
	res := make([]byte, 0)
	for i := 0; i < len(b); i++ {
		if b[i] == ' ' {
			res = append(res, []byte("%20")...)
		} else {
			res = append(res, b[i])
		}
	}
	return string(res)
}